### DISCUSSION TITLE
recovery service

### DISCUSSION THREAD

**Discussion by rcasta74** [davmac314/dinit#371] [Q&A] [answered]
Hi,
after a power outage the boot fails and I am usually asked to take an action since all services were stopped without a shutdown command having been issued (I supposed this is the recovery service); a reboot usually works and let the system to start.
I'd like to use dinit on an headless server: is there a way to configure the recovery service to automatically reboot the system instead of to wait for a choice?
Thanks

**Comment by davmac314** [comment, marked as answer]
This might be more a question for your distribution (Artix?).

You can edit the service description file for the `recovery` service just like you can any other service, and by that means you can make it do whatever you want, including running `/sbin/reboot` for example (maybe you'd need `--use-passed-cfd` argument to reboot, and `options = pass-cs-fd` in the service description, as otherwise the control socket might not get created as part of a failed boot).

However, the recovery service is only supposed to run when the system _can't_ boot, and in that case rebooting wouldn't help. It seems like there is currently an issue in Artix where the filesystem checks incorrectly fail boot if they correct an error, that is discussed here:

https://forum.artixlinux.org/index.php/topic,7030.msg43766.html#msg43766

I think that is the real issue here.

Note that, though you can change the `recovery` service as I mentioned above, the interactive prompt after boot failure is not part of the service itself and can't currently be disabled.

**Comment by rcasta74** [comment]
Ok, I missed to check into /lib/dinit.d/ folder; because of this I didn't found file descriptor for recovery service and I was thinking it was an internal service to dinit.

> the interactive prompt after boot failure is not part of the service itself and can't currently be disabled.

Ok, I missed that as well, so recovery service wasn't even started. I tryed to simulate power outage a couple of time, but both times the system started without issues, so I cannot investigate deeper what happened; I can just remember that after the fsck program recovered journal on all FS, all following services failed to start. Trying to choice `restart` instead of `reboot` I remember that the services failed again and in the output I have seen somewere that all FS were mounted read-only; because of this a simple reboot then did the trick.

Thanks for the link on artix forum, I will look at it.
